.github/workflows/ci.yml

```diff
@@ -103,9 +103,67 @@ jobs:
       - name: Install node modules
         run: yarn install --frozen-lockfile
       - name: Run integration tests
-        run: yarn integration-tests\
+        run: yarn integration-tests
       # TODO: Set up slack notifications
       # - name: Running size integration tests (failures are reported in Slack only).
       #   run: |
       # If the size integration tests fail, report the failure to a dedicated #components-ci-size-tracking Slack channel.
       # yarn integration-tests:size-test || yarn ci-notify-slack-failure components-ci-size-tracking
+
+  linker_aot_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Initialize environment
+        uses: angular/dev-infra/github-actions/npm/checkout-and-setup-node@9931e1a8d1b62fcd2267e89f9993a494856cc1cd
+        with:
+          cache-node-modules: true
+      - name: Setup Bazel
+        uses: angular/dev-infra/github-actions/bazel/setup@9931e1a8d1b62fcd2267e89f9993a494856cc1cd
+      - name: Setup Bazel RBE
+        uses: angular/dev-infra/github-actions/bazel/configure-remote@9931e1a8d1b62fcd2267e89f9993a494856cc1cd
+      - name: Install node modules
+        run: yarn install --frozen-lockfile
+      - name: Run linker AOT tests
+        run: yarn test-linker-aot
+      # TODO: Set up slack notifications
+      # - name: Running size integration tests (failures are reported in Slack only).
+      #   run: |
+      # If the size integration tests fail, report the failure to a dedicated #components-ci-size-tracking Slack channel.
+      # yarn integration-tests:size-test || yarn ci-notify-slack-failure components-ci-size-tracking
+
+  linker_jit_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Initialize environment
+        uses: angular/dev-infra/github-actions/npm/checkout-and-setup-node@9931e1a8d1b62fcd2267e89f9993a494856cc1cd
+        with:
+          cache-node-modules: true
+      - name: Setup Bazel
+        uses: angular/dev-infra/github-actions/bazel/setup@9931e1a8d1b62fcd2267e89f9993a494856cc1cd
+      - name: Setup Bazel RBE
+        uses: angular/dev-infra/github-actions/bazel/configure-remote@9931e1a8d1b62fcd2267e89f9993a494856cc1cd
+      - name: Install node modules
+        run: yarn install --frozen-lockfile
+      - name: Run linker JIT tests
+        run: yarn test-linker-jit
+      # TODO: Set up slack notifications
+      # - name: Running size integration tests (failures are reported in Slack only).
+      #   run: |
+      # If the size integration tests fail, report the failure to a dedicated #components-ci-size-tracking Slack channel.
+      # yarn integration-tests:size-test || yarn ci-notify-slack-failure components-ci-size-tracking
+
+  test:
+    runs-on: ubuntu-latest-4core
```

Comment:
I think we are hitting a point where we should discuss/audit that we are not regressing with our CI times- especially given the lower spec'ed host VMs compared to CircleCI. Do we have any data here?

Reply:
I don't have anything, but I would prefer that we start low and find we need to up them instead of just jumping to big runners right away. 

Unfortunately I haven't been able to find a good way to "benchmark" to be able to say what we should use.

Reply:
Sorry for being a little pushy about this. I just think it's important to keep CI times similar to before. Historically Angular components had the fastest wall time of CI across all repositories and I feel strongly that we want to keep this. We should not simply switch to GitHub actions, use significantly different runners without auditing/comparing. 

Even if we are saying we can revisit in the future- in practice we will likely never do this. We are not immediately jumping to big runners- but we should audit/compare and understand the impact on the Bazel host etc. We are just looking at not regressing in CI wall time. Additionally smaller runners might incur the same cost as larger runners which take fewer minutes.

Reply:
Also capturing this here (not related to runner size I assume), I'm seeing a 10min time for lint, while on CircleCI it took 2min. See: https://github.com/angular/components/actions/runs/6034665778/job/16373509530

Reply:
I did some testing and found that 16core is the sweet spot of speed/cost.

Reply:
Perfect. Thanks for looking more.